### PR TITLE
[FIX] Sanitize inputs

### LINKED ISSUE #684: EPI coregistration issues and confusing errors
https://openneuro.org/datasets/ds001219/versions/00001?app=MRIQC&version=61&job=0a22ab9c-2673-4da8-80e4-19cf87d7a41e

```
2018-02-06 22:38:06,444 niworkflows:INFO Estimating initial transform using AffineInitializer
2018-02-06 22:38:07,036 niworkflows:INFO Terminal outputs of initialization saved (/usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPI2MNI/stderr.nipype-init).
2018-02-06 22:38:07,036 niworkflows:INFO Loading settings from file /usr/local/miniconda/lib/python3.6/site-packages/niworkflows/data/epi-mni_registration_precise_000.json.
2018-02-06 22:38:07,039 niworkflows:INFO Retry #0, commandline: 
antsRegistration --collapse-output-transforms 1 --dimensionality 3 --float 0 --initial-moving-transform [ /usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPI2MNI/transform.mat, 0 ] --initialize-transforms-per-stage 0 --interpolation LanczosWindowedSinc --output [ epi_to_mni, epi_to_mni_Warped.nii.gz ] --transform Rigid[ 0.05 ] --metric Mattes[ /usr/local/miniconda/lib/python3.6/site-packages/mriqc/data/mni/2mm_T2_brain.nii.gz, /usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPIApplyMask/sub-04_task-main_run-01_bold_volreg_tstat_corrected_masked.nii.gz, 1, 56, Regular, 0.25 ] --convergence [ 10000x1000x100, 1e-06, 20 ] --smoothing-sigmas 4.0x2.0x1.0vox --shrink-factors 4x2x1 --use-estimate-learning-rate-once 1 --use-histogram-matching 1 --transform Affine[ 0.08 ] --metric Mattes[ /usr/local/miniconda/lib/python3.6/site-packages/mriqc/data/mni/2mm_T2_brain.nii.gz, /usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPIApplyMask/sub-04_task-main_run-01_bold_volreg_tstat_corrected_masked.nii.gz, 1, 56, Regular, 0.25 ] --convergence [ 500x250x100, 1e-06, 20 ] --smoothing-sigmas 4.0x2.0x1.0vox --shrink-factors 8x4x2 --use-estimate-learning-rate-once 1 --use-histogram-matching 1 --transform SyN[ 0.1, 3.0, 0.0 ] --metric CC[ /usr/local/miniconda/lib/python3.6/site-packages/mriqc/data/mni/2mm_T2_brain.nii.gz, /usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPIApplyMask/sub-04_task-main_run-01_bold_volreg_tstat_corrected_masked.nii.gz, 1, 4, None, 1 ] --convergence [ 100x30x20, 1e-06, 10 ] --smoothing-sigmas 3.0x2.0x1.0vox --shrink-factors 8x4x2 --use-estimate-learning-rate-once 1 --use-histogram-matching 1 --winsorize-image-intensities [ 0.005, 0.995 ]  --write-composite-transform 1
2018-02-06 22:38:08,302 niworkflows:WARNING Retry #0 failed: Command:
antsRegistration --collapse-output-transforms 1 --dimensionality 3 --float 0 --initial-moving-transform [ /usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPI2MNI/transform.mat, 0 ] --initialize-transforms-per-stage 0 --interpolation LanczosWindowedSinc --output [ epi_to_mni, epi_to_mni_Warped.nii.gz ] --transform Rigid[ 0.05 ] --metric Mattes[ /usr/local/miniconda/lib/python3.6/site-packages/mriqc/data/mni/2mm_T2_brain.nii.gz, /usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPIApplyMask/sub-04_task-main_run-01_bold_volreg_tstat_corrected_masked.nii.gz, 1, 56, Regular, 0.25 ] --convergence [ 10000x1000x100, 1e-06, 20 ] --smoothing-sigmas 4.0x2.0x1.0vox --shrink-factors 4x2x1 --use-estimate-learning-rate-once 1 --use-histogram-matching 1 --transform Affine[ 0.08 ] --metric Mattes[ /usr/local/miniconda/lib/python3.6/site-packages/mriqc/data/mni/2mm_T2_brain.nii.gz, /usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPIApplyMask/sub-04_task-main_run-01_bold_volreg_tstat_corrected_masked.nii.gz, 1, 56, Regular, 0.25 ] --convergence [ 500x250x100, 1e-06, 20 ] --smoothing-sigmas 4.0x2.0x1.0vox --shrink-factors 8x4x2 --use-estimate-learning-rate-once 1 --use-histogram-matching 1 --transform SyN[ 0.1, 3.0, 0.0 ] --metric CC[ /usr/local/miniconda/lib/python3.6/site-packages/mriqc/data/mni/2mm_T2_brain.nii.gz, /usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPIApplyMask/sub-04_task-main_run-01_bold_volreg_tstat_corrected_masked.nii.gz, 1, 4, None, 1 ] --convergence [ 100x30x20, 1e-06, 10 ] --smoothing-sigmas 3.0x2.0x1.0vox --shrink-factors 8x4x2 --use-estimate-learning-rate-once 1 --use-histogram-matching 1 --winsorize-image-intensities [ 0.005, 0.995 ]  --write-composite-transform 1
Standard output:

Standard error:

Return code: 1.
2018-02-06 22:38:08,303 workflow:WARNING [Node] Error on "workflow_enumerator.funcMRIQC.SpatialNormalization.EPI2MNI" (/usr/local/src/mriqc/work/workflow_enumerator/funcMRIQC/SpatialNormalization/_in_file_..snapshot..data..sub-04..func..sub-04_task-main_run-01_bold.nii.gz/EPI2MNI)
2018-02-06 22:38:08,358 workflow:ERROR Node EPI2MNI.a0 failed to run on host 42bf9a034ab7.
2018-02-06 22:38:08,360 workflow:ERROR Saving crash info to /output/data/logs/crash-20180206-223808-root-EPI2MNI.a0-99097ca4-20e7-4141-8b33-ccfa1d606682.txt
Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 62, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 443, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 520, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 596, in _run_command
    result = self._interface.run()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base/core.py", line 485, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/interfaces/report_base.py", line 54, in _run_interface
    ReportCapableInterface, self)._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/interfaces/mni.py", line 147, in _run_interface
    term_out = _write_outputs(interface_result.runtime,
AttributeError: 'NoneType' object has no attribute 'runtime'
```

I don't see any issues with the image
image

### PR DESCRIPTION
**Pull request by chrisfilo**
Closes #684

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 numpy>=1.12.0
-niworkflows>=0.2.7
+git+https://github.com/chrisfilo/niworkflows.git@21c0647bb7e42938e947334ab8c474c011fe3484#egg=niworkflows-0.3.1-dev
```
Comment (oesteban) [inline requirements.txt]: I guess you are planning on merging poldracklab/niworkflows#221 and release first, so the requirements will finally be pinned to that new release, right?

Comment (chrisfilo) [reply requirements.txt]: Yes - that's why this PR is still marked as WIP.

**Comment by chrisfilo** [comment]
I don't know the MRIQC reporting framework well enough to assess how hard this would be. I can at least open an issue for this.

**Comment by chrisfilo** [comment]
Ready to merge.

**Comment by oesteban** [comment]
Could you roll back the latest commit? Since MRIQC has the automated script to update the changelog, this is not necessary anymore.

(supposing that your origin remote points to your fork of mriqc):

```Bash
git reset --hard HEAD^
git push origin +fix/sanitize_inputs
```

**Comment by chrisfilo** [comment]
done

**Pull request merged**